### PR TITLE
[BatchRuns] Improve err and default strategy

### PR DESCRIPTION
**Pull request by liranbg**
Improving the validation and error message when batch run provide with wrong combination of strategy and its param file.
e.g., running strategy list with param file of json is invalid because it essentially means the strategy yields matrix combinations and not list of combinations (e.g.: cross between each vertical).

https://jira.iguazeng.com/browse/ML-4449

### CODE REVIEW

File: tests/runtimes/test_generators.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+            mlrun.runtimes.generators.ListGenerator,
+            does_not_raise(),
+        ),
+        # no strategy, default to grid
+        (
+            "",
+            "hyperparams.json",
+            mlrun.runtimes.generators.GridGenerator,
+            does_not_raise(),
+        ),
```
Comment (theSaarco) [inline tests/runtimes/test_generators.py:58]: Add a case for `list` with the `JSON` file (should work, produce 2 iterations).

**Review by theSaarco — CHANGES_REQUESTED** [review]
Error message should be improved. Also some minor improvements to the test.

File: tests/runtimes/test_generators.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    )
+
+    with expected_error:
+        generator = mlrun.runtimes.generators.get_generator(run_spec, execution, None)
+        assert isinstance(
+            generator, expected_generator_class
+        ), f"unexpected generator type {type(generator)}"
+
+        if strategy == "list":
+            assert generator.df.keys().to_list() == ["p1", "p2"]
```
[the remark this answers is not in the document]
Comment (liranbg) [reply tests/runtimes/test_generators.py:110]: done

File: tests/runtimes/test_generators.py (hunk 1 above)
Comment (liranbg) [reply tests/runtimes/test_generators.py:71]: done

**Review by theSaarco — APPROVED** [review]
Gr8

**Pull request merged**